zetaclient/chains/solana/signer/signer.go

```diff
@@ -0,0 +1,467 @@
+	// FIXME: config this; right now it's the same privkey used by local e2e test_solana_*.go
+	privkey := solana.MustPrivateKeyFromBase58(
+		"4yqSQxDeTBvn86BuxcN5jmZb2gaobFXrBqu8kiE9rZxNkVMe3LfXmFigRsU4sRp7vk4vVP1ZCFiejDKiXBNWvs2C",
+	)
+
+	// create withdraw instruction with program call data
+	var err error
+	var inst solana.GenericInstruction
+	inst.DataBytes, err = borsh.Serialize(contract.WithdrawInstructionParams{
+		Discriminator: contract.DiscriminatorWithdraw(),
+		Amount:        msg.Amount,
+		Signature:     msg.SigRS(),
+		RecoveryID:    msg.SigV(),
+		MessageHash:   msg.Hash(),
+		Nonce:         msg.Nonce,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot serialize withdraw instruction")
+	}
+
+	fmt.Printf("tx withdraw message: %d, %d, %d, %s\n", msg.ChainID, msg.Nonce, msg.Amount, msg.To.String())
+
+	// attach required accounts to the instruction
+	var accountSlice []*solana.AccountMeta
+	accountSlice = append(accountSlice, solana.Meta(privkey.PublicKey()).WRITE().SIGNER())
+	accountSlice = append(accountSlice, solana.Meta(signer.pda).WRITE())
+	accountSlice = append(accountSlice, solana.Meta(msg.To).WRITE())
+	accountSlice = append(accountSlice, solana.Meta(signer.gatewayID))
+	inst.ProgID = signer.gatewayID
+	inst.AccountValues = accountSlice
+
+	// get a recent blockhash
+	recent, err := signer.client.GetRecentBlockhash(ctx, rpc.CommitmentFinalized)
+	if err != nil {
+		return nil, errors.Wrap(err, "GetRecentBlockhash error")
+	}
+
+	// create a transaction that wraps the instruction
+	tx, err := solana.NewTransaction(
+		[]solana.Instruction{&inst},
+		recent.Value.Blockhash,
+		solana.TransactionPayer(privkey.PublicKey()),
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "NewTransaction error")
+	}
+
+	// fee payer signs the transaction
+	_, err = tx.Sign(func(key solana.PublicKey) *solana.PrivateKey {
+		if key.Equals(privkey.PublicKey()) {
+			return &privkey
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "signer unable to sign transaction")
+	}
+
+	return tx, nil
+}
+
+// TryProcessOutbound - signer interface implementation
+// This function will attempt to build and sign a Solana transaction using the TSS signer.
+// It will then broadcast the signed transaction to the Solana chain.
+func (signer *Signer) TryProcessOutbound(
+	ctx context.Context,
+	cctx *types.CrossChainTx,
+	outboundProc *outboundprocessor.Processor,
+	outboundID string,
+	_ interfaces.ChainObserver,
+	zetacoreClient interfaces.ZetacoreClient,
+	height uint64,
+) {
+	// end outbound process on panic
+	defer func() {
+		outboundProc.EndTryProcess(outboundID)
+		if err := recover(); err != nil {
+			signer.Logger().Std.Error().Msgf("TryProcessOutbound: %s, caught panic error: %v", cctx.Index, err)
+		}
+	}()
+
+	// prepare logger
+	params := cctx.GetCurrentOutboundParam()
+	logger := signer.Logger().Std.With().
+		Str("OutboundID", outboundID).
```

Comment:
```suggestion
		Str("outbound.id", outboundID).
```

Reply:
Not sure what's the difference would be. In Datadog, we can search log thru
`@SendHash:0x8ca7b6b0685b34a59b419af00e673b737f0bed0716fcf1c6c101f71afcbf656a`
or
`@OutboundID:0x8ca7b6b0685b34a59b419af00e673b737f0bed0716fcf1c6c101f71afcbf656a-1-29045`

Does the additional `.` produce a sub filter in Datadog?

From what I've seen so far, there are a few common fields can be used for signer (or outbound) log print. I'll unify
the signer logger as below. The `method`, `chain`, `nonce` and `cctx` are pretty important meta data for troubleshooting when incidents happen. Here are some filters in DD

![image](https://github.com/user-attachments/assets/83449290-492e-4e05-9582-c280ce57977e)

```
logger := signer.Logger().Std.With().
    Str("method", "TryProcessOutbound").
    Int64("chain", signer.Chain().ChainId).
    Uint64("nonce", params.TssNonce).
    Str("cctx", cctx.Index).
    Logger()
```